c2/c2-client-bundle/c2-client-service/src/test/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandlerTest.java

```diff
@@ -0,0 +1,234 @@
+        assertEquals(DEBUG, testHandler.getOperandType());
+    }
+
+    @Test
+    public void testC2CallbackUrlIsNullInArgs() {
+        // given
+        DebugOperationHandler testHandler = DebugOperationHandler.create(c2Client, VALID_BUNDLE_FILE_LIST, DEFAULT_CONTENT_FILTER);
+        C2Operation c2Operation = operation(null);
+
+        // when
+        C2OperationAck result = testHandler.handle(c2Operation);
+
+        // then
+        assertEquals(OPERATION_ID, result.getOperationId());
+        assertEquals(NOT_APPLIED, result.getOperationState().getState());
+    }
+
+    @Test
+    public void testFilesAreCollectedAndUploadedAsATarGzBundle() {
+        // given
+        Map<String, String> bundleFileNamesWithContents = asList("file.log", "application.conf", "default.properties")
+            .stream()
+            .collect(toMap(identity(), __ -> DEFAULT_FILE_CONTENT));
+        List<Path> createBundleFiles = bundleFileNamesWithContents.entrySet().stream()
+            .map(entry -> placeFileWithContent(entry.getKey(), entry.getValue()))
+            .collect(toList());
+        DebugOperationHandler testHandler = DebugOperationHandler.create(c2Client, createBundleFiles, DEFAULT_CONTENT_FILTER);
+        C2Operation c2Operation = operation(C2_DEBUG_UPLOAD_ENDPOINT);
+
+        // when
+        C2OperationAck result = testHandler.handle(c2Operation);
+
+        // then
+        ArgumentCaptor<String> uploadUrlCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<byte[]> uploadBundleCaptor = ArgumentCaptor.forClass(byte[].class);
+        verify(c2Client).uploadDebugBundle(uploadUrlCaptor.capture(), uploadBundleCaptor.capture());
+        assertEquals(OPERATION_ID, result.getOperationId());
+        assertEquals(FULLY_APPLIED, result.getOperationState().getState());
+        assertEquals(C2_DEBUG_UPLOAD_ENDPOINT, uploadUrlCaptor.getValue());
+        Map<String, String> resultBundle = extractBundle(uploadBundleCaptor.getValue());
+        assertTrue(mapEqual(bundleFileNamesWithContents, resultBundle));
+    }
+
+    @Test
+    public void testFileToCollectDoesNotExist() {
+        // given
+        DebugOperationHandler testHandler = DebugOperationHandler.create(c2Client, singletonList(Paths.get(tempDir.getAbsolutePath(), "missing_file")), DEFAULT_CONTENT_FILTER);
+        C2Operation c2Operation = operation(C2_DEBUG_UPLOAD_ENDPOINT);
+
+        // when
+        C2OperationAck result = testHandler.handle(c2Operation);
+
+        // then
+        assertEquals(OPERATION_ID, result.getOperationId());
+        assertEquals(NOT_APPLIED, result.getOperationState().getState());
+    }
+
+    @Test
+    public void testContentIsFilteredOut() {
+        // given
+        String filterKeyword = "minifi";
+        String bundleFileName = "filter_content.file";
+        String fileContent = Stream.of("line one", "line two " + filterKeyword, filterKeyword + "line three", "line four").collect(joining(NEW_LINE));
```

Comment:
What do you think about extending with a case when the keyword is inside the string, just to be safe :)
Also validating what happens if for some reason everything / nothing is filtered out. I know super edge case but could be interesting.

Reply:
Extended test case

Reply:
Thanks for extending it